### ISSUE TITLE
 PHP Fatal error:  Uncaught TypeError: abs(): Argument #1 ($num) must be of type int|float, string given

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#31570]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I get this error hundreds of times in the Apache error log:
> Got error 'PHP message: PHP Fatal error:  Uncaught TypeError: abs(): Argument #1 ($num) must be of type int|float, string given in controllers/front/CartController.php:79

It comes from an incoorect URL (I don't know yet why it is called) `https://example.com/panier?add=1&id_product=41261&qty=`.
`panier` certainly is `cart`in an english shop. Notice the missing `qty=` value at the end.

### Expected behavior

I expect the code to validate parameters, in order to avoid server errors when a wrong URL is called.

### Steps to reproduce

1. Navigate in the FO to the page `https://example.com/panier?add=1&id_product=41261&qty=` in a French shop (or `[https://example.com/cart?add=1&id_product=41261&qty=` in an English shop)
2. Look into the apache error log. You will notice an error like above. The error is not visible in `BO/Advanced Parameters/Logs` nor in `var/logs`.

### PrestaShop version(s) where the bug happened

8.0.1

### PHP version(s) where the bug happened

8.0

**Comment by kpodemski** [comment]
I confirm the issue. `abs` requires `int|float`, and passing the second parameter to `Tools::getValue` is causing an issue; thus, it's better to have it typed.

**Comment by boherm** [comment]
I'm on it.

In fact, first of all we have some issue with our `Tools::getValue` when we need a default value. 
When we go to `/cart?...&qty=` qty is set but empty and default value isn't set as value, but if we omit `qty` in request default value is set.

And with `abs` function:
```<?php
var_dump(abs(12));	// int(12)
var_dump(abs("12"));	// int(12)
var_dump(abs(false));	// int(0)
var_dump(abs(""));	// PHP 7.4: int(0)
			// PHP 8.X: Fatal error: must be of type int|float...
```

I will fix that this afternoon.

**Comment by kpodemski** [comment]
@boherm https://github.com/PrestaShop/PrestaShop/pull/31571 👍🏻

**Comment by joecoolio66** [comment]
I'm not understanding if this has been fixed or is in development... we're seeing this error on 8.0.4 with php 8.1.  Can someone please comment?  Thanks

**Comment by kpodemski** [comment]
This fix fill be available in PrestaShop 9, but you can apply the changes today:
https://github.com/PrestaShop/PrestaShop/pull/31571/files

**Comment by joecoolio66** [comment]
Thanks!  I assume this will have to be changed after each PS upgrade until the fix is part of the release.

**Comment by kpodemski** [comment]
Yes. Maybe we will backport this fix to 8.1.x, but not to 8.0.x, for sure.
